client/state/reader/follows/test/reducer.js

```diff
@@ -486,11 +490,118 @@ describe( 'reducer', () => {
 				type: READER_FOLLOW_ERROR,
 				payload: { feedUrl: 'http://discoverinvalid.wordpress.com', error: 'invalid_feed' },
 			} );
-			expect( state[ 'discoverinvalid.wordpress.com' ] ).to.eql( {
+			chaiExpect( state[ 'discoverinvalid.wordpress.com' ] ).to.eql( {
 				is_following: true,
 				error: 'invalid_feed',
 			} );
 		} );
+
+		test( 'should add a new follow from a site request', () => {
+			const original = deepFreeze( {} );
+			const incomingSite = {
+				ID: 125,
+				name: 'Postcards from the Reader',
+				URL: 'https://postcardsfromthereader.wordpress.com',
+				feed_URL: 'https://postcardsfromthereader.wordpress.com',
+				is_following: true,
+				subscription: {
+					delivery_methods: {
+						email: {
+							send_posts: true,
+							send_comments: false,
+							post_delivery_frequency: 'weekly',
+						},
+					},
+				},
+			};
+
+			const state = items( original, {
+				type: READER_SITE_REQUEST_SUCCESS,
+				payload: incomingSite,
+			} );
+
+			expect( state[ 'postcardsfromthereader.wordpress.com' ] ).toEqual( {
+				is_following: true,
+				blog_ID: 125,
+				URL: 'https://postcardsfromthereader.wordpress.com',
+				feed_URL: 'https://postcardsfromthereader.wordpress.com',
+				delivery_methods: {
+					email: {
+						send_posts: true,
+						send_comments: false,
+						post_delivery_frequency: 'weekly',
+					},
+				},
+			} );
+		} );
+
+		test( 'should update an existing follow from a site request', () => {
+			const original = deepFreeze( {
+				is_following: true,
+				blog_ID: 125,
+				URL: 'https://postcardsfromthereader.wordpress.com',
+				feed_URL: 'https://postcardsfromthereader.wordpress.com',
+				delivery_methods: {
+					email: {
+						send_posts: false,
+						send_comments: true,
+					},
+				},
+			} );
+			const incomingSite = {
+				ID: 125,
+				name: 'Postcards from the Reader',
+				URL: 'https://postcardsfromthereader.wordpress.com',
+				feed_URL: 'https://postcardsfromthereader.wordpress.com',
+				is_following: true,
+				subscription: {
+					delivery_methods: {
+						email: {
+							send_posts: true,
+							send_comments: false,
+							post_delivery_frequency: 'weekly',
+						},
+					},
+				},
+			};
+
+			const state = items( original, {
+				type: READER_SITE_REQUEST_SUCCESS,
+				payload: incomingSite,
+			} );
+
+			expect( state[ 'postcardsfromthereader.wordpress.com' ] ).toEqual( {
+				is_following: true,
+				blog_ID: 125,
+				URL: 'https://postcardsfromthereader.wordpress.com',
+				feed_URL: 'https://postcardsfromthereader.wordpress.com',
+				delivery_methods: {
+					email: {
+						send_posts: true,
+						send_comments: false,
+						post_delivery_frequency: 'weekly',
+					},
+				},
+			} );
+		} );
+
+		test( 'should disregard an unfollowed site from a site request', () => {
```

Comment:
if the incoming site is not followed, are you sure we should throw it away?
Maybe we should accept it and turn email settings to false right etc etc

Reply:
can we get an unfollowed site that has email settings? by the db schema it's possible, but i'm not sure if the api is preventing it from flowing out...

Reply:
I suppose there might be a case to accept it, even if it only updates `is_following` to `false`.

I'll check how the API handles the unfollowed case.

Reply:
It looks like it is technically possible to have email delivery settings but not be following the site in Reader. We try and create the corresponding subscription record wherever possible but there are orphaned email subscriptions in existence. 

I think it's probably best to reflect the email subscription settings if they exist, because the user will be receiving emails from the subscription even if they're not following the site.

I'll alter the reducer to stop throwing away sites where `is_followed` is false.